### ISSUE TITLE
Add an option to persist data-bind attributes

### ISSUE THREAD

**Issue by gustavnikolaj** [Munawwar/htmlizer#18]
I would like a feature that allows a data-bind to be applied, but also persisted.

It has some drawbacks- in particular regarding inlined templates - but it is something that will enable us to easily bootstrap parts of a knockout application server side to help non-js users and crawlers.

**Comment by Munawwar** [comment]
Question: Are there also bindings within your data-bind attribute that are exclusively for the client-side?

**Comment by gustavnikolaj** [comment]
Yes. There probably will be. I'd prefer if htmlizer, when put in this mode, would just ignore bindings that it does not support.

```js
var template = '<div data-bind="text: textContent, foo: bar"></div>';
var viewModel = { textContent: 'foobar' };
var output = (new HTMLizer(template).toString(viewModel);

/* output should contain the following string:
 * <div data-bind="text: textContent, foo: bar">foobar</div>
 */
```

This might be problematic as the `bar` value in the `foo` binding might trigger a ReferenceError when you evaluate the binding value. But this is the behaviour that I want...

**Comment by gustavnikolaj** [comment]
If it prooves to be way to complex, I could live with having to write a transform function that took a template of 
```html
<div data-bind="text: textContent, foo: bar"></div>
```
and converted it into
```html
<div data-htmlizer="text: textContent" data-bind="text: textContent, foo: bar"></div>
```

Essentially it's just a matter of duplicating the `data-bind` attribute and filtering the bindings down to the subset that htmlizer supports.

**Comment by Munawwar** [comment]
It is easy to ignore bindings that isn't supported.

What I was concerned is whether you provide data for all the supported bindings. If in your last example, textContent is undefined, then it would be more difficult to figure out when to ignore the binding and when to take undefined as a valid input. In KO and Htmlizer, null and undefined is valid input for the text binding (it just converts them to empty string). So hopefully that isn't part of your use case.

**Comment by gustavnikolaj** [comment]
Anything rendered on the server should be discarded once it re-renders on the client.

The main caveat of that is that we will have to avoid anonymous templates.

If we have this template:

```html
```

**Comment by Munawwar** [comment]
@gustavnikolaj : Have a look whether the changes in the keepbindings branch satisfies your use case.

One more feature Htmlizer is missing at the moment is KO custom elements/components. Maybe you'd need that as well to write "isomorphic KO".

**Comment by gustavnikolaj** [comment]
Thanks alot! I'll make sure to get back to you once I've tried it out :-)

**Comment by gustavnikolaj** [comment]
@Munawwar LGTM! :+1:

**Comment by Munawwar** [comment]
Merged and released v2.1.0.
